base/reflection.jl

```diff
@@ -224,15 +219,40 @@ function _methods(t::Array,i,lim::Integer,matching::Array{Any,1})
     return matching
 end
 
-function methods(f::ANY)
-    ft = typeof(f)
-    if ft <: Type || !isempty(ft.parameters)
-        # for these types of `f`, not every method in the table will necessarily
-        # match, so we need to filter based on its type.
-        return methods(f, Tuple{Vararg{Any}})
-    else
-        return ft.name.mt
+# high-level, more convenient method lookup functions
+
+# type for reflecting and pretty-printing a subset of methods
+type MethodSet
+    ms::Array{Method,1}
+    mt::MethodTable
+end
+
+length(m::MethodSet) = length(m.ms)
+isempty(m::MethodSet) = isempty(m.ms)
+start(m::MethodSet) = start(m.ms)
+done(m::MethodSet, s) = done(m.ms, s)
+next(m::MethodSet, s) = next(m.ms, s)
+getindex(m::MethodSet, i::Int) = m.ms[i]
```

Comment:
I'll remove `getindex`, which will give us more flexibility in the future. How about `MethodList`, since it is list-like in supporting forward iteration but not getindex?